### PR TITLE
Correction for Throwable, phpdoc, add missing attributes

### PR DESCRIPTION
**Pull request by SQKo**
This PR tries to add exception where possible, as well correcting existing throwable and its phpdoc

Based on this https://www.php.net/manual/en/spl.exceptions.php#spl.exceptions.tree the closest exception should be used, for example the command builder had `RangeException` while it should be `LengthException` to adhere the `LogicException` where the code must be fixed at compile time and not `RuntimeException` where the code may have error during run time, due to command builder is being written at compile time.

A `RunTimeException` inside `Promise` of a `Http` request should be also changed to `reject()` instead of `throw`

Also for checking permission if creating request is allowed or not.

Currently have completed with Interactions (Commands and Message Components), will check other classes later.

Also added missing attributes from several Part classes, the phpdoc property attributes is sorted following the API Documentation, and the fillable is sorted by the JSON payload data

### CODE REVIEW

**Comment by SQKo** [comment]
PR is almost done, just need testing after reviews... (Don't merge it yet after approval)

**Review by valzargaming — APPROVED** [review]

**Review by valzargaming — APPROVED** [review]

**Review by SQKo — COMMENTED** [review]
I've reviewed and did quick runtime test, and can be merged if there are no more problems.

**Review by valzargaming — APPROVED** [review]

**Pull request merged**
